aea/mail/base.py

```diff
@@ -612,12 +610,10 @@ def _stop(self):
 
             except asyncio.CancelledError:
                 logger.debug("Receiving loop cancelled.")
-                for t in task_to_connection.keys():
-                    await t
-                return
+                break
```

Comment:
these two break make the execution flow to go outside the while loop, iterate over all the open tasks (see https://github.com/fetchai/agents-aea/pull/610/files#diff-8c7ccf06c0c0d3423bd6701658db0823R619-R620) and cancel them.